robot/rospackages/src/mcu_control/scripts/ArmNode.py

```diff
@@ -4,101 +4,86 @@
 import traceback
 import time
 import re
-
-from robot.rospackages.src.mcu_control.scripts.SerialUtil import init_serial, get_serial
+import serial
 
 import rospy
 from std_msgs.msg import String, Header, Float32
 from sensor_msgs.msg import JointState
 from mcu_control.srv import *
 
-mcuName = 'arm'
-
-# todo: test ros+website over network with teensy
-# todo: make a MCU serial class that holds the port initialization stuff and returns a reference?
-# todo: put similar comments and adjustments to code in the publisher and server demo scrips once finalized
-requests = {
-    'ping' : ['pong'],
-    'who' : ['arm'],
-    'loop open' : ['open'],
-    'loop closed' : ['closed'],
-    'armspeed' : ['Alert', 'Success'],
-    'reboot' : ['rebooting']
-}
-
-def handle_client(req):
-    ser = get_serial()
-    global reqFeedback
-    global reqInWaiting
-    armResponse = ArmRequestResponse()
-    timeout = 0.1 # 100ms timeout
-    reqInWaiting=True
-    sinceRequest = time.time()
-    rospy.loginfo('received '+req.msg+' request from GUI, sending to arm Teensy')
-    ser.write(str.encode(req.msg+'\n')) # ping the teensy
-    while armResponse.success is False and (time.time()-sinceRequest < timeout):
-        if reqFeedback is not '':
-            print(reqFeedback)
-            for request in requests:
-                for response in requests[request]:
-                    if request == req.msg and response in reqFeedback:
-                        armResponse.response = reqFeedback
-                        armResponse.success = True #a valid request and a valid response from the
-                        break
-            if armResponse.success:
-                break
-            else:
-                armResponse.response += reqFeedback
-        rospy.Rate(100).sleep()
-    rospy.loginfo('took '+str(time.time()-sinceRequest)+' seconds, sending this back to GUI: ')
-    rospy.loginfo(armResponse)
-    reqFeedback=''
-    reqInWaiting=False
-    return armResponse
+
+def handle_pong(data):
+    print("Received", data.decode('utf-8'))
+
+def handle_get_angles(data):
+    print("Received", data.join(" "))
+    anglePub.publish(data) # todo: convert each value to the correct type
+
+# https://docs.google.com/spreadsheets/d/1bE3h0ZCqPAUhW6Gn6G0fKEoOPdopGTZnmmWK1VuVurI/edit#gid=1131090349
+out_commands = [("estop", 0), ] # todo add all commands...
+in_commands = [("ping", 16, handle_pong), ("get_angles", 17, handle_get_angles)]
+
+
+def get_handler(commandId):
+    for in_command in in_commands:
+        if commandId == in_command[1]:
+            return in_command[2]
+    return None
+
+# Pin definitions
+arm_pin = 11
+wheel_pin = 13
+science_pin = 15
+
+teensy_pins = [arm_pin, wheel_pin, science_pin]
+
+ser = serial.Serial('/dev/ttyACM0', 57600) # you sure this is good for the jetson tim?
```

Comment:
we're gonna need a separate node that handles traffic cause this way there are gonna be collisions. 
For testing it's OK as long as you only have this one node running